api/src/main/java/com/google/sps/SendNotifications.java

```diff
@@ -20,98 +16,156 @@
 import com.google.api.services.gmail.Gmail;
 
 import java.io.ByteArrayOutputStream;
+import java.io.File;
 import java.io.IOException;
+import java.net.URL;
 import java.util.Properties;
+import java.util.concurrent.ExecutionException;
 import java.util.ArrayList;
 import java.util.Arrays;
 import java.util.List;
 import java.util.Map;
 
 import javax.mail.MessagingException;
+import javax.mail.Multipart;
 import javax.mail.Session;
+import javax.mail.Transport;
 import javax.mail.internet.*;
 
 import org.apache.commons.codec.binary.Base64;
-import org.springframework.boot.autoconfigure.SpringBootApplication;
-import org.springframework.web.bind.annotation.GetMapping;
 import org.springframework.web.bind.annotation.PostMapping;
 import org.springframework.web.bind.annotation.RestController;
 
 @RestController
 public class SendNotifications {
+  boolean firebaseInitialized = false;
 
-  public List<Email> gatherRecipients() throws IOException {
-    GoogleCredentials credentials = GoogleCredentials.getApplicationDefault();
+  public List<EmailCreation> gatherRecipients(HttpServletRequest request, HttpServletResponse response, Firestore db)
+      throws IOException {
 
-    System.out.println("Google Credentials created...");
-
-    String projectId = "phoebeliang-step";
-    FirebaseOptions options = new FirebaseOptions.Builder().setCredentials(credentials).setProjectId(projectId).build();
-    FirebaseApp.initializeApp(options);
-
-    System.out.println("App Initialized...");
-
-    Firestore db = FirestoreClient.getFirestore();
-
-    String gameID = "dT6U1wQKR1rGJcEncyMS";
-    List<Email> emails = new ArrayList<>();
+    String gameID = request.getParameter("gameID");
+    String emailType = request.getParameter("emailType");
```

Comment:
Consider converting the email type to an enum using valueOf for cheaper and safer comparisons https://www.tutorialspoint.com/java/lang/enum_valueof.htm

Advantages of enums (and use case examples): http://www.thefinestartist.com/effective-java/30

Reply:
Haven't used these much, but I believe I implemented correctly. Will push soon so you can check it out